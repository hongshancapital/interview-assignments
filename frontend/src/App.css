.App {
  text-align: center;
  position: relative;
}
