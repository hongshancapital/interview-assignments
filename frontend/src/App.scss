.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}
