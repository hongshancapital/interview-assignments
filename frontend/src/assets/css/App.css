.App {
  text-align: center;
  position: relative;
}


/*slides*/
.color {
  mix-blend-mode: difference;
  color: #fff;
}

.slides {
  width: 500px;
  height: 300px;
  overflow: hidden;
  margin: auto;
  position: relative;
}

.slides-item {
  display: flex;
  width: 100%;
  height: 100%;
  transition: transform 1s;
}

.slide {
  position: relative;
  width: 100%;
  background-color: #000;
}

.slide-img {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  object-fit: cover;
}

.slide-title {
  font-size: 24px;
  font-weight: 600;
  white-space: pre-wrap;
  margin: 20px;
}

.slide-text {
  padding: 0 20px;
  margin: 0;
  font-size: 18px;
}

.slides-nav {
  position: absolute;
  bottom: 10px;
  left: 0;
  width: 100%;
  display: flex;
  justify-content: center;
}

.nav-bar {
  width: 30px;
  height: 3px;
  background-color: rgba(0, 0, 0, 0.5);
  margin-right: 5px;
  border-radius: 20px;
  overflow: hidden;
}

.nav-progress {
  background-color: #fff;
  background-blend-mode: difference;
  width: 0;
  height: 100%;
}