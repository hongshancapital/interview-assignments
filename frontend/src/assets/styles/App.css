.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}

.my_carousel_item {
  width: 480px;
  height: 300px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: relative;
}

.my_carousel_item2 {
  width: 960px;
  height: 600px;
}

.my_carousel_item3 {
  width: 1440px;
  height: 900px;
}

.my_carousel_item .description {
  position: absolute;
  text-align: center;
  width: 100%;
  top: 36px;
}

.my_carousel_item._white {
  color: #fff;
}

.my_carousel_item h3 {
  font-size: 26px;
}
.my_carousel_item p {
  font-size: 12px;
}

.my_carousel_item2 h3 {
  font-size: 56px;
}
.my_carousel_item2 p {
  font-size: 28px;
}

.my_carousel_item3 h3 {
  font-size: 86px;
}
.my_carousel_item3 p {
  font-size: 43px;
}
