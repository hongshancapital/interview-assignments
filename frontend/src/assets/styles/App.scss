.App {
  text-align: center;
  position: relative;
}

.title {
  font-size: 56px;
  font-weight: 600;
}

.text {
  font-size: 28px;
  font-weight: 400;
}

.my_carousel_item {
  width: 480px;
  height: 300px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: relative;
  &.my_carousel_item2 {
    width: 960px;
    height: 600px;
    h3 {
      font-size: 56px;
    }
    p {
      font-size: 28px;
    }
  }
  &.my_carousel_item3 {
    width: 1440px;
    height: 900px;
    h3 {
      font-size: 86px;
    }
    p {
      font-size: 43px;
    }
  }

  &._white {
    color: #fff;
  }
  .description {
    position: absolute;
    text-align: center;
    width: 100%;
    top: 36px;
  }

  h3 {
    font-size: 26px;
  }
  p {
    font-size: 12px;
  }
}
