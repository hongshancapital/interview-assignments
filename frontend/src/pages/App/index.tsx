import React from 'react'
import './index.scss'
import Carousel from '../../components/Carousel'
import CarouselItem, { ICarouselItemProps } from '../../components/CarouselItem'
import { mockData } from '../../model'

function App() {
  return (
    <div className="App">
      <Carousel autoplay={true} autoPlayInterval={3000}>
        {mockData.map((item: ICarouselItemProps) => {
          return <CarouselItem {...item} key={item.key} />
        })}
      </Carousel>
    </div>
  )
}

export default App
