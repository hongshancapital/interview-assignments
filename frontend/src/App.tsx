import React from "react";
import "./App.css";
import Carousel from "./Carousel";

function App() {
  return (
    <div className="App">
      <Carousel />
    </div>
  );
}

export default App;
