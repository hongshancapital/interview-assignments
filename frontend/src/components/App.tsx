import React from "react";
import Carousel from "./Carousel";
import { SLIDES_DATA, DELAY_TIME } from "../common/constants";
import "../assets/css/App.css";

function App() {
  return (
  	<div className="App">
  		<Carousel slides={SLIDES_DATA} delay={DELAY_TIME} />
  	</div>
  );
}

export default App;
