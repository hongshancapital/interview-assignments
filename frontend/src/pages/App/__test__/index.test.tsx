import React from 'react'
import { render } from '@testing-library/react'
import App from '../index'

describe('Test App Page', () => {
  test('App Page Render Snapshot', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
