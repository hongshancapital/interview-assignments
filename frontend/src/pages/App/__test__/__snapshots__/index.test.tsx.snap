// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Page App Page Render Snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="com-carousel-wrap"
    >
      <div
        class="com-carousel com-offset0"
      >
        <div
          class="com-carousel-item-box prefix-carousel-item-wrap prefix-carousel-item-dark"
        >
          <img
            class="prefix-carousel-item-image"
            src="iphone.png"
          />
          <div
            class="prefix-carousel-item"
          >
            <h2
              class="prefix-carousel-item-header"
            >
              xPhone
            </h2>
            <p
              class="prefix-carousel-item-content"
            >
              Lots to love. Less to spend.
            </p>
            <p
              class="prefix-carousel-item-content"
            >
              Starting at $399
            </p>
          </div>
        </div>
        <div
          class="com-carousel-item-box prefix-carousel-item-wrap prefix-carousel-item-light"
        >
          <img
            class="prefix-carousel-item-image"
            src="tablet.png"
          />
          <div
            class="prefix-carousel-item"
          >
            <h2
              class="prefix-carousel-item-header"
            >
              Tablet
            </h2>
            <p
              class="prefix-carousel-item-content"
            >
              Just the right amount of everything
            </p>
          </div>
        </div>
        <div
          class="com-carousel-item-box prefix-carousel-item-wrap prefix-carousel-item-gray"
        >
          <img
            class="prefix-carousel-item-image"
            src="airpods.png"
          />
          <div
            class="prefix-carousel-item"
          >
            <h2
              class="prefix-carousel-item-header"
            >
              Buy a Tablet or xPhone for college.
            </h2>
            <h2
              class="prefix-carousel-item-header"
            >
              Get airPods
            </h2>
          </div>
        </div>
      </div>
      <ul
        class="com-dots"
      >
        <li
          class="com-dots-item"
        >
          <div
            class="com-dots-inner com-dots-actived"
            style="transition-duration: 3000ms;"
          />
        </li>
        <li
          class="com-dots-item"
        >
          <div
            class="com-dots-inner"
            style="transition-duration: 0ms;"
          />
        </li>
        <li
          class="com-dots-item"
        >
          <div
            class="com-dots-inner"
            style="transition-duration: 0ms;"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
